openai_adapter.py

```
@@ -0,0 +1,265 @@
+            "type": "object",
+            "properties": {},
+            "required": []
+        }
+        
+        for name, param in parameters:
+            # Add to properties
+            param_type = "string"  # Default type
+            
+            # Try to determine the type from the annotation
+            if param.annotation != inspect.Parameter.empty:
+                if param.annotation == str:
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)